### PR TITLE
Mac friendly configuration

### PR DESCRIPTION
**Pull request by jrper**
A collection of changes to configure etc. to make building on Mac OSX easier. These come in two flavours:
1) Changes required due to a BSD-based environment in which the gnu toolchain may be harder to find, for which there are variations in compiler flags
2) Changes to deal with the possibility of a case-insensitive file system.

There are also two changes to tests proposed, one to deal with problems accessing lib spud from inside the embedded python interpreter and a tolerance relaxation to one of @cianwilson's tests.

@tmbgreaves Could I get a buildbot queue for this as well please?

### CODE REVIEW

**Comment by tmbgreaves** [comment]
Done:

http://buildbot.ese.ic.ac.uk:8080/builders/mac_friendly

**Comment by stephankramer** [comment]
The changes look all very sensible to me. Only the changes to configure.in look to have some indentation issue. Does python libspud not work, or only when called from within fluidity?

In any case, if someone with a mac system could try this out independently? - then I'm happy for this to go in.

**Comment by jrper** [comment]
The indentation issues are from meld again, sorry, I'll fix it up before the merge into master. Libspud works fine for me from a stand alone python interpreter, but causes segfaults when loaded from the embedded python in fluidity when compiled in a Mac framework build. If anyone with a Mac does want to try it then I'm very happy to assist regarding installing the supporting software libraries, and would be especially interested to talk to anyone who uses Homebrew.

File: tests/diffusion_2d_p0_adaptive_parallel/diffusion_2d_p0_adaptive_parallel.xml [hunk 1]
```diff
@@ -1,51 +1,32 @@
-<?xml version="1.0" encoding="UTF-8" ?>
```
Comment (stephankramer) [inline tests/diffusion_2d_p0_adaptive_parallel/diffusion_2d_p0_adaptive_parallel.xml]: I was wondering what the hell is going on with this file. But it looks like this was a hand-crafted .xml originally and this is what happens if you open and save it in diamond.

File: configure.in [hunk 2]
```diff
@@ -80,6 +80,25 @@ FLFEMDEM="femdem"
 # Check system type
 AC_CANONICAL_HOST
 
+case $host_os in
+     darwin* ) 
+	     LINUX=no
```
Comment (stephankramer) [inline configure.in]: One last indentation issue - this is to proof I actually read the diff again ;-)

**Comment by stephankramer** [comment]
All looks good!

**Comment by tmbgreaves** [comment]
Buildbot running again: http://buildbot.ese.ic.ac.uk:8080/builders/mac_friendly

**Pull request merged**